test/gleam/map_test.gleam

```diff
@@ -189,13 +208,131 @@ pub fn fold_test() {
   |> map.fold(0, add)
   |> should.equal(6)
 
-  let concat = fn(acc, k, _) { string.append(acc, k) }
+  let prepend = fn(acc, k, _) { list.prepend(acc, k) }
 
   dict
-  |> map.fold("", concat)
-  |> should.equal("abcd")
+  |> map.fold([], prepend)
+  |> list.sort(string.compare)
+  |> should.equal(["a", "b", "c", "d"])
 
   map.from_list([])
   |> map.fold(0, add)
   |> should.equal(0)
 }
+
+fn range(start, end, a) {
+  case end - start {
+    n if n < 1 -> a
+    _ -> range(start, end - 1, [end - 1, ..a])
+  }
+}
```

Comment:
I think this is `list.range`

Reply:
list.range blows up the callstack atm, and there was also talk of changing inclusive/exclusive end values (not sure if that change is in yet). This will always do what I want :)